extensions/test/win/echo_extension/XWalkExtension.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+
+namespace xwalk
+{
+    public class XWalkExtension
+    {
+      public XWalkExtension()
+      {}
+
+      public String ExtensionName()
+      {
+        return "echo";
+      }
+
+      public String ExtensionAPI()
+      {
+        return
+          @"var echoListener = null;
+          extension.setMessageListener(function(msg) {
+            if (echoListener instanceof Function) {
+              echoListener(msg);
+            };
+          });
+          exports.echo = function(msg, callback) {
+            echoListener = callback;
+            extension.postMessage(msg);
+          };
+          exports.syncEcho = function(msg) {
+            return extension.internal.sendSyncMessage(msg);
+          };";
+      }
+
+    }
+}
```

Comment:
According the very little .cs files in chromium, seems we need follow the C++ coding style for C# code.